packages/vite/src/node/optimizer/scan.ts

```diff
@@ -498,6 +513,29 @@ function esbuildScanPlugin(
         { filter: htmlTypesRE, namespace: 'file' },
         htmlTypeOnLoadCallback,
       )
+      // css
+      build.onLoad(
+        { filter: CSS_LANGS_RE, namespace: 'file' },
+        globImportOnLoadCallback,
+      )
+      // json & wasm
+      build.onLoad(
+        { filter: /\.(json|json5|wasm)$/, namespace: 'file' },
+        globImportOnLoadCallback,
+      )
+      // known asset types
+      build.onLoad(
+        {
+          filter: new RegExp(`\\.(${KNOWN_ASSET_TYPES.join('|')})$`),
+          namespace: 'file',
+        },
+        globImportOnLoadCallback,
+      )
+      // known vite query types: ?worker, ?raw
+      build.onLoad(
+        { filter: SPECIAL_QUERY_RE, namespace: 'file' },
+        globImportOnLoadCallback,
+      )
```

Comment:
Would you make a function that calls both `build.onResolve` and `build.onLoad` so that we don't forget to change the other one?

Reply:
@XiSenao Sorry, I mislead you. I mean something like

```js
const setupExternalize = (filter: RegExp, doExternalize: (path: string) => boolean) => {
  build.onResolve({ filter }, ({ path }) => {
    if(doExternalize(path)) return { path, external: true }
  })
  // add comment here about the esbuild's issue
  build.onLoad({ filter, namespace: 'file' }, () => {
    // TODO
  })
}


```

Reply:
Sorry, I misunderstood. 😂